### DISCUSSION TITLE
[Chapter 12] 낮은 결합도가 어플리케이션에 미치는 영향?

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#200] [Chapter 12. Spring Events] [answered]
해당 챕터는 스프링 이벤트를 설명하고 사용방법을 알려줬던 챕터입니다.
책을 읽어보면, 이벤트 기반의 로직처리가 어플리케이션에게 주는 장점이 있다고 설명합니다.

그 중 느슨한 결합(낮은 결합도)는 어플리케이션 설계단계에서 중요하게 다뤄지는 개념이라고 생각합니다.

느슨한 결합도가 주는 장점과 결합도가 높은 설계가 가진 한계점과 단점에 대해 한번 논의해봤으면 좋겠습니다.

**Comment by JoisFe** [comment, marked as answer]
결합도는 낮게 응집도는 높게 구성하는 프로그램의 장점으로는
해당 코드를 앞으로 유지보수 할때 얼마나 편리한가? 를 정하는 것 같습니다.
한 클래스가 다른 클래스와 결합이 높다면 다른 클래스의 변동이 생기면 기존 클래스의 기능에 문제가 생길 수 있어 유지보수 하는데 힘들어지기 때문입니다.
만약 여러 클래스와 결합이 생긴다면 더욱 유지보수가 힘들어지게 되고 그만큼 기능 수정 혹은 추가를 함에 있어 비용이 계속해서 많이 들게 된다고 생각합니다.
